helm/templates/logger-network-policy.yaml

```diff
@@ -18,15 +20,29 @@ spec:
     ports:
     - protocol: TCP
       port: 8500
-  {{- if not $.Values.minio.metricsLogger.withRedis }}
+  {{- if or (not $.Values.minio.metricsLogger.withRedis) (eq $.Values.minio.metricsLogger.LOG_PROVIDER "elasticsearch") (eq $.Values.minio.metricsLogger.LOG_PROVIDER "s3") }}
   egress:
   - to:
     - ipBlock:
         cidr: 0.0.0.0/0
     ports:
     - protocol: UDP
       port: 53
+    {{- if not $.Values.minio.metricsLogger.withRedis }}
     - protocol: TCP
       port: {{ $.Values.minio.metricsLogger.REDIS_PORT }}
+    {{- end }}
+    {{- if (eq $.Values.minio.metricsLogger.LOG_PROVIDER "elasticsearch") }}
+    - protocol: TCP
+      port: {{ $.Values.minio.metricsLogger.ES_PORT }}
+    {{- else if (eq $.Values.minio.metricsLogger.LOG_PROVIDER "s3") }}
+    {{- if $.Values.minio.metricsLogger.S3_NON_AWS_TARGET }}
+    - protocol: TCP
+      port: {{ (split ":" $.Values.minio.metricsLogger.S3_ENDPOINT)._2 }}
```

Comment:
the S3 endpoint might not include the port number, in which case it should be parsed according to the scheme (http:// or https://)